### PR TITLE
Add new postgres shared server configuration

### PR DESCRIPTION
**Pull request by TannerFilip**
let it be known that i fixed a huge mess of commits by using `git push --force`

### CODE REVIEW

File: db.tf [hunk 1]
```diff
@@ -68,3 +69,37 @@ resource "aws_route53_record" "mysql-shared-dns" {
...
+    final_snapshot_identifier = "postgres-shared-db-final"
+    name                      = "postgresshareddb"
+    username                  = "root"
+    password                  = "${var.postgres-shared-db_password}"
+    vpc_security_group_ids    = ["${aws_security_group.shared-rds-sg.id}"]
+    db_subnet_group_name      = "${aws_db_subnet_group.apps-shared-rds-subnetgroup.name}"
+    parameter_group_name      = "defaults.postgres9.4"
+    tags {
+        Name = "postgres-shared-db"
+        app = "mysql"
```
Comment (flamingspaz) [inline db.tf]: This should be postgres

File: db.tf [hunk 2]
```diff
@@ -68,3 +69,37 @@ resource "aws_route53_record" "mysql-shared-dns" {
   ttl = 300
   records = ["${aws_db_instance.mysql-shared-db.address}"]
 }
+
+resource "aws_db_instance" "postgres-shared-db" {
+    allocated_storage         = 300
+    engine                    = "postgres"
+    instance_class            = "db.m4.xlarge"
+    publicly_accessible       = "false"
```
Comment (flamingspaz) [inline db.tf]: This should be a bool and not a string

File: db.tf [hunk 3]
```diff
@@ -68,3 +69,37 @@ resource "aws_route53_record" "mysql-shared-dns" {
   ttl = 300
   records = ["${aws_db_instance.mysql-shared-db.address}"]
 }
+
+resource "aws_db_instance" "postgres-shared-db" {
+    allocated_storage         = 300
+    engine                    = "postgres"
```
Comment (flamingspaz) [inline db.tf]: We should also specify `engine_version` so we know which version of postgres we're going to get. Amazon recently launched support for 9.5 so we should go with the latest version of that (9.5.4)

File: db.tf [hunk 4]
```diff
@@ -68,3 +69,37 @@ resource "aws_route53_record" "mysql-shared-dns" {
...
+    apply_immediately         = true
+    multi_az                  = true
+    storage_type              = "gp2"
+    final_snapshot_identifier = "postgres-shared-db-final"
+    name                      = "postgresshareddb"
+    username                  = "root"
+    password                  = "${var.postgres-shared-db_password}"
+    vpc_security_group_ids    = ["${aws_security_group.shared-rds-sg.id}"]
+    db_subnet_group_name      = "${aws_db_subnet_group.apps-shared-rds-subnetgroup.name}"
+    parameter_group_name      = "defaults.postgres9.4"
```
Comment (flamingspaz) [inline db.tf]: This should just be `default`, and see above about postgres 9.5.

File: db.tf [hunk 5]
```diff
@@ -68,3 +69,37 @@ resource "aws_route53_record" "mysql-shared-dns" {
   ttl = 300
   records = ["${aws_db_instance.mysql-shared-db.address}"]
 }
+
+resource "aws_db_instance" "postgres-shared-db" {
```
Comment (flamingspaz) [inline db.tf]: Something I forgot to do with the MySQL instance was give it an `identifier`, which is the name that shows up in the console. Use `postgres-shared-db` as with the terraform resource name.

**Comment by flamingspaz** [comment]
test this please

**Comment by flamingspaz** [comment]
test this please

File: db.tf [hunk 6]
```diff
@@ -68,3 +69,39 @@ resource "aws_route53_record" "mysql-shared-dns" {
...
+    identifier                = "postgres-shared-db"
+    allocated_storage         = 300
```
Comment (johngian) [inline db.tf]: Please scale that down. 300G is *too* much.

File: db.tf [hunk 7]
```diff
@@ -68,3 +69,39 @@ resource "aws_route53_record" "mysql-shared-dns" {
...
+    vpc_security_group_ids    = ["${aws_security_group.shared-rds-sg.id}"]
+    db_subnet_group_name      = "${aws_db_subnet_group.apps-shared-rds-subnetgroup.name}"
+    parameter_group_name      = "default.postgres9.5"
+    tags {
+        Name = "postgres-shared-db"
+        app = "postgres"
+        env = "shared"
+        project = "partinfra"
+    }
+
```
Comment (johngian) [inline db.tf]: Please remove the trailing lines here.

File: db.tf [hunk 8]
```diff
@@ -68,3 +69,39 @@ resource "aws_route53_record" "mysql-shared-dns" {
...
+    multi_az                  = true
+    storage_type              = "gp2"
+    final_snapshot_identifier = "postgres-shared-db-final"
+    name                      = "postgresshareddb"
+    username                  = "root"
+    password                  = "${var.postgres-shared-db_password}"
+    vpc_security_group_ids    = ["${aws_security_group.shared-rds-sg.id}"]
+    db_subnet_group_name      = "${aws_db_subnet_group.apps-shared-rds-subnetgroup.name}"
+    parameter_group_name      = "default.postgres9.5"
+    tags {
```
Comment (johngian) [inline db.tf]: Please add an extra line here to make it more readable.

**Comment by johngian** [comment]
It probably makes sense to enable ``copy_tags_to_snapshot``.

**Comment by johngian** [comment]
Please change the commit message to 

```
[Bug XXXXX] Add postgres shared server configuration
```

**Comment by johngian** [comment]
Test this please

**Comment by johngian** [comment]
ok to test

**Pull request merged**